packages/core/src/components/forms/numericInput.tsx

```diff
@@ -0,0 +1,407 @@
+     * @default false
+     */
+    disabled?: boolean;
+
+    /** The name of icon (the part after `pt-icon-`) to render on left side of input */
+    leftIconName?: string;
+
+    /** The placeholder text in the absence of any value */
+    placeholder?: string;
+
+    /**
+     * The increment between successive values when `shift` is held
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** The maximum value of the input */
+    max?: number;
+
+    /** The minimum value of the input */
+    min?: number;
+
+    /**
+     * The increment between successive values when `alt` is held
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * The increment between successive values when no modifier keys are held
+     * @default 1
+     */
+    stepSize?: number;
+
+    /** The value to display in the input field */
+    value?: number | string;
+
+    /** The callback invoked when `enter` is pressed and when the field loses focus */
+    onConfirm?(value: string): void;
+
+    /** The callback invoked when the value changes */
+    onUpdate?(value: string): void;
+}
+
+export interface INumericInputState {
+    isInputGroupFocused?: boolean;
+    isButtonGroupFocused?: boolean;
+    shouldSelectAfterUpdate?: boolean;
+    value?: string;
+}
+
+@PureRender
+export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInputProps, INumericInputState> {
+    public static displayName = "Blueprint.NumericInput";
+
+    public static defaultProps: INumericInputProps = {
+        buttonPosition: Position.RIGHT,
+        majorStepSize: 10,
+        minorStepSize: 0.1,
+        stepSize: 1,
+        value: NumericInput.VALUE_EMPTY,
+    };
+
+    private static DECREMENT_KEY = "decrement";
+    private static INCREMENT_KEY = "increment";
+    private static DECREMENT_ICON_NAME = "chevron-down";
+    private static INCREMENT_ICON_NAME = "chevron-up";
+    private static VALUE_EMPTY = "";
+    private static VALUE_ZERO = "0";
+
+    private inputElement: HTMLInputElement;
+
+    public constructor(props?: HTMLInputProps & INumericInputProps) {
+        super(props);
+        this.state = {
+            shouldSelectAfterUpdate: false,
+            value: this.getValueOrEmptyValue(props),
+        };
+    }
+
+    public render() {
+        const { buttonPosition, className } = this.props;
+
+        const inputGroup = (
+            <InputGroup
+                {...removeNonHTMLProps(this.props)}
+                intent={this.props.intent}
+                inputRef={this.inputRef}
+                key="input-group"
+                leftIconName={this.props.leftIconName}
+                onFocus={this.handleInputFocus}
+                onBlur={this.handleInputBlur}
+                onChange={this.handleInputUpdate}
+                onKeyDown={this.handleInputKeyDown}
+                value={this.state.value}
+            />
+        );
+
+        if (buttonPosition === "none" || buttonPosition === null) {
```

Comment:
This is intentional:
- `=== undefined` is equivalent to `"right"`, the default value
- `=== null` is equivalent to `"none"`

I'll add a comment.